test/test_tensorboard.py

```diff
@@ -0,0 +1,454 @@
+precision = [0.3333333, 0.3786982, 0.5384616, 1.0, 0.0]
+recall = [1.0, 0.8533334, 0.28, 0.0666667, 0.0]
+
+
+class WriterTest(unittest.TestCase):
+    def test_writer(self):
+        with SummaryWriter() as writer:
+            sample_rate = 44100
+
+            n_iter = 0
+            writer.add_scalar('data/scalar_systemtime', 0.1, n_iter)
+            writer.add_scalar('data/scalar_customtime', 0.2, n_iter, walltime=n_iter)
+            writer.add_scalars('data/scalar_group', {"xsinx": n_iter * np.sin(n_iter),
+                                                     "xcosx": n_iter * np.cos(n_iter),
+                                                     "arctanx": np.arctan(n_iter)}, n_iter)
+            x = np.zeros((32, 3, 64, 64))  # output from network
+            writer.add_images('Image', x, n_iter)  # Tensor
+            writer.add_image_with_boxes('imagebox',
+                                        np.zeros((3, 64, 64)),
+                                        np.array([[10, 10, 40, 40], [40, 40, 60, 60]]),
+                                        n_iter)
+            x = np.zeros(sample_rate * 2)
+
+            writer.add_audio('myAudio', x, n_iter)
+            writer.add_video('myVideo', np.random.rand(16, 48, 1, 28, 28).astype(np.float32), n_iter)
+            # writer.add_text('Text', 'text logged at step:' + str(n_iter), n_iter)
+            # writer.add_text('markdown Text', '''a|b\n-|-\nc|d''', n_iter)
+            writer.add_histogram('hist', np.random.rand(100, 100), n_iter)
+            # writer.add_pr_curve('xoxo', np.random.randint(2, size=100), np.random.rand(
+            #     100), n_iter)  # needs tensorboard 0.4RC or later
+            # writer.add_pr_curve_raw('prcurve with raw data', true_positive_counts,
+            #                         false_positive_counts,
+            #                         true_negative_counts,
+            #                         false_negative_counts,
+            #                         precision,
+            #                         recall, n_iter)
+            # export scalar data to JSON for external processing
+            writer.export_scalars_to_json("./all_scalars.json")
+
+
+def test_linting():
+    import subprocess
+    subprocess.check_output(['flake8', 'tensorboardX'])
+
+
+class SummaryWriterTest(unittest.TestCase):
+    def test_summary_writer_ctx(self):
+        # after using a SummaryWriter as a ctx it should be closed
+        with SummaryWriter(filename_suffix='.test') as writer:
+            writer.add_scalar('test', 1)
+        assert writer.file_writer is None
+
+    def test_summary_writer_close(self):
+        # Opening and closing SummaryWriter a lot should not run into
+        # OSError: [Errno 24] Too many open files
+        passed = True
+        try:
+            writer = SummaryWriter()
+            writer.close()
+        except OSError:
+            passed = False
+
+        assert passed
+
+    def test_windowsPath(self):
```

Comment:
Can you change this to  `def test_windows_path(self):`?

Reply:
Also, is this a test that should only be run on Windows? Honestly, I think you can probably remove this test.

Reply:
removed